syntax = "proto3";


message Address {
  string ip = 1;
  uint32 port = 2;
}

message GatewayLocation {
  Address address = 1;
}

message DeviceInfo {
  string name = 1;
  string state = 2;
  string metadata = 3;
}

message JoinRequest {
  DeviceInfo device_info = 1;
  Address device_address = 2;
}

message JoinReply {
  uint32 report_interval = 1;
}
